framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java

```diff
@@ -505,6 +506,31 @@ public void typeProcess(TypeElement element, TreePath tree) {
         }
     }
 
+    /**
+     * Issues a warning about any {@code @SuppressWarnings} that isn't used by this checker, but
+     * contains a key that would suppress a warning from this checker.
+     */
+    @Override
+    protected void warnUnneededSuppressions() {
+        if (parentChecker != null || !hasOption("warnUnneededSuppressions")) {
+            return;
+        }
+        Set<Element> elementsSuppress = new HashSet<>(this.elementsWithSuppressedWarnings);
+        this.elementsWithSuppressedWarnings.clear();
+        Set<String> checkerKeys = new HashSet<>(getSuppressWarningsKeys());
+        Set<String> errorKeys = new HashSet<>(messages.stringPropertyNames());
```

Comment:
It would be nice if there were less duplication between the two methods.
It seems important that this method needs to call `this.elementsWithSuppressedWarnings.clear();`. Slightly restructuring to ensure this would be best, either by adding another function or calling `super`. Alternatively, document this requirement.